### PR TITLE
Build Dashboard with TypeScript 2.0

### LINKED ISSUE #13160: [Dashboard] Upgrade TypeScript to version 2.0.x
### Description

Currently Dashboard is built with TypeScript 1.8. It's better be upgraded for me by some reasons.

* I want to use `new URL(string)` in my patch but `lib.d.ts` in TypeScript 1.8 has been broken. (refs https://github.com/Microsoft/TypeScript/issues/2583 )
* Referring to [Beaking-Changes](https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#typescript-20), breaking changes between 1.8 and 2.0 will be a compiler detective.
* Newer will be better. :)

**OS and version:**    

The master branch, commit id 8636720d

### PR DESCRIPTION
**Pull request by monaka**
### What does this PR do?

Upgrades TypeScript for building Dashboard to 2.0. And some components reported by `npm audit` also. 

### What issues does this PR fix or reference?

fixes #13160. refs #13147.

### CODE REVIEW

File: dashboard/package.json [hunk 1]
```diff
@@ -100,7 +103,7 @@
     "angular-animate": "1.4.8",
     "angular-aria": "1.4.8",
     "angular-cookies": "1.4.8",
-    "angular-file-upload": "2.0.0",
+    "angular-file-upload": "^2.5.0",
```
Comment (benoitf) [inline dashboard/package.json:108]: FYI dependencies need Eclipse CQ to be updated

File: dashboard/package.json [hunk 2]
```diff
@@ -115,9 +118,11 @@
     "angular-websocket": "1.0.9",
     "bootstrap-styl": "4.0.5",
     "codemirror": "5.19.0",
+    "eslint": "^1.0.0",
```
Comment (benoitf) [inline dashboard/package.json]: IMHO these dependencies are dev dependencies (jasmine-core as well)

Comment (monaka) [reply dashboard/package.json]: Right. I'll fix them.

File: dashboard/package.json (hunk 1 above)
Comment (monaka) [reply dashboard/package.json:108]: Thank you for your information. I didn't know about CQ and I'm browsing [IPZilla](https://dev.eclipse.org/ipzilla/buglist.cgi?query_format=advanced&short_desc_type=allwordssubstr&short_desc=TypeScript&long_desc_type=substring&long_desc=&bug_file_loc_type=allwordssubstr&bug_file_loc=&keywords_type=allwords&keywords=&emailassigned_to1=1&emailtype1=substring&email1=&emailassigned_to2=1&emailreporter2=1&emailcc2=1&emailtype2=substring&email2=&bugidtype=include&bug_id=&chfieldfrom=&chfieldto=Now&chfieldvalue=&cmdtype=doit&order=Importance&field0-0-0=noop&type0-0-0=noop&value0-0-0=) now.
Hmm... I see it's better to upgrade in step-by-step (First, just upgrade CQ passed only. Next, apply CQs) approach.

**Comment by monaka** [comment]
I close this for now as this won't be Eclipse CQ approved. I'll send the another PR.

**Pull request closed without merging**